remoteworkitem/trackerquery_repository.go

```diff
@@ -28,29 +28,29 @@ func NewTrackerQueryRepository(db *gorm.DB) *GormTrackerQueryRepository {
 	return &GormTrackerQueryRepository{db}
 }
 
+// TrackerQueryRepository encapsulate storage & retrieval of tracker queries
+type TrackerQueryRepository interface {
+	repository.Exister
+	Create(ctx context.Context, query string, schedule string, tracker uuid.UUID, spaceID uuid.UUID) (*app.TrackerQuery, error)
+	Save(ctx context.Context, tq app.TrackerQuery) (*app.TrackerQuery, error)
+	Load(ctx context.Context, ID string) (*app.TrackerQuery, error)
+	Delete(ctx context.Context, ID string) error
+	List(ctx context.Context) ([]*app.TrackerQuery, error)
+}
+
 // Create creates a new tracker query in the repository
 // returns BadParameterError, ConversionError or InternalError
-func (r *GormTrackerQueryRepository) Create(ctx context.Context, query string, schedule string, tracker string, spaceID uuid.UUID) (*app.TrackerQuery, error) {
-	tid, err := strconv.ParseUint(tracker, 10, 64)
-	if err != nil || tid == 0 {
-		// treating this as a not found error: the fact that we're using number internal is implementation detail
-		return nil, NotFoundError{"tracker", tracker}
-	}
-
-	log.Info(ctx, map[string]interface{}{
-		"tracker_id": tid,
-	}, "Tracker ID to be created")
-
+func (r *GormTrackerQueryRepository) Create(ctx context.Context, query string, schedule string, tracker uuid.UUID, spaceID uuid.UUID) (*app.TrackerQuery, error) {
 	tq := TrackerQuery{
 		Query:     query,
 		Schedule:  schedule,
-		TrackerID: tid,
+		TrackerID: tracker,
 		SpaceID:   spaceID,
 	}
 	tx := r.db
 	if err := tx.Create(&tq).Error; err != nil {
 		log.Error(ctx, map[string]interface{}{
-			"tracker_id":    tid,
+			"tracker_id":    tracker.String(),
```

Comment:
please log `err` here and other locations in the file as well.

Reply:
fixed it here -> 
71b44b5